service/src/main/java/skills/storage/repos/UserPointsRepo.groovy

```diff
@@ -660,9 +661,9 @@ interface UserPointsRepo extends CrudRepository<UserPoints, Integer> {
                 (lower(CONCAT(ua.first_name, ' ', ua.last_name, ' (',  ua.user_id_for_display, ')')) like lower(CONCAT(\'%\', ?3, \'%\'))  OR
                  lower(ua.user_id_for_display) like lower(CONCAT('%', ?3, '%'))
                 ) and 
-                up.skill_id is null 
-            GROUP BY up.user_id''', nativeQuery = true)
-    List<ProjectUser> findDistinctProjectUsersAndUserIdLike(String projectId, String usersTableAdditionalUserTagKey, String query, Pageable pageable)
+                up.skill_id is null
+            GROUP BY up.user_id) AS projectUser WHERE projectUser.totalPoints >= ?4''', nativeQuery = true)
```

Comment:
interestingly the way user's project points are represented in ``user_points`` table  ``totalPoints >= ?4`` filter can be directly applied to the where clause of the initial query (`up.totalPoints >= ?4`); this is because the data guarantees that only 1 `user_points` row per user is returned via ``up.project_id=?1 and up.skill_id is null`` and that row represents user's total points for that project; for a larger data-set this is a very taxing and slow(ish) query and probably worth not adding an extra selection layer;  

The same applies to other `find` queries and `count` queries